ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/deduplication/CommandDeduplicationBase.scala

```diff
@@ -477,39 +554,118 @@ private[testtool] abstract class CommandDeduplicationBase(
           selfServiceErrorCode = LedgerApiErrors.ConsistencyErrors.DuplicateCommand,
           exceptionMessageSubstring = None,
           checkDefiniteAnswerMetadata = true,
+          assertDeduplicatedSubmissionIdAndOffsetOnError(
+            request.commands.map(_.submissionId),
+            acceptedOffset,
+            _,
+          ),
         )
       )
-  }
 
-  protected def submitRequestAndAssertAsyncDeduplication(ledger: ParticipantTestContext)(
+  protected def submitRequestAndAssertAsyncDeduplication(
+      ledger: ParticipantTestContext,
       request: SubmitRequest,
+      acceptedOffset: LedgerOffset,
       parties: Party*
   )(implicit ec: ExecutionContext): Future[OffsetWithCompletion] =
-    submitRequestAndAssertCompletionStatus(
+    submitRequestAndAssertCompletion(
       ledger
-    )(request, Code.ALREADY_EXISTS, parties: _*)
+    )(
+      request,
+      parties: _*
+    ) { completion =>
+      assertCompletionStatus(request, completion, Code.ALREADY_EXISTS)
+      assertDeduplicatedSubmissionIdAndOffsetOnCompletion(
+        request.commands.map(_.submissionId),
```

Comment:
The submission id here should be from the request which generated the `acceptedoffset`

Reply:
Hopefully c30f3fd; btw, it looks to me like we were reusing submission IDs as we were not generating new requests, is that possible?

Reply:
Hmm, that should not happen afaik. 
From what I see we always set a new submission id for the submit request. Even if we wouldn't have set that, it should've been randomly generated.

Reply:
Yep, found where it's done, thanks

Reply:
Should be fixed.